src/node_i18n.cc

```diff
@@ -468,6 +468,21 @@ int32_t ToUnicode(MaybeStackBuffer<char>* buf,
   // #46. Therefore, explicitly filters out that error here.
   info.errors &= ~UIDNA_ERROR_EMPTY_LABEL;
 
+  // These error conditions are mandated unconditionally by UTS #46 version
+  // 9.0.0 (rev. 17), but were found to be incompatible with many actual domain
+  // names in the wild. As such, in the current UTS #46 draft (rev. 18) these
+  // checks are made optional depending on the CheckHyphens flag, which will be
+  // disabled in WHATWG URL's "domain to unicode" algorithm as soon as the UTS
+  // #46 draft becomes standard.
+  // Refs:
+  // - https://github.com/whatwg/url/issues/53
+  // - http://www.unicode.org/review/pri317/
+  // - http://www.unicode.org/reports/tr46/tr46-18.html
+  // - https://www.icann.org/news/announcement-2000-01-07-en
+  info.errors &= ~UIDNA_ERROR_HYPHEN_3_4;
+  info.errors &= ~UIDNA_ERROR_LEADING_HYPHEN;
+  info.errors &= ~UIDNA_ERROR_TRAILING_HYPHEN;
+
```

Comment:
Unfortunately I was [mistaken](https://github.com/nodejs/node/pull/12966#issuecomment-301385482) in my understanding of ToUnicode, which means that the entire block (ll. 464-485) shouldn't be necessary. While at it, can you delete this block, the erroneous `|| (!lenient && info.errors != 0)` condition below, as well as the now-unnecessary `lenient` parameter?